### DISCUSSION TITLE
How to get active branch in a script?

### DISCUSSION THREAD

**Discussion by RedLine89** [netboxlabs/netbox-branching#154] [Q&A] [answered]
Hey,

I am exploring branching capabilities in scripting can't seem to figure out how to get an active branch selected by a user.. The code below seem correct (as far as my limited knowledge of NetBox goes), but I always get `False` even when any of the branches in the list is actually selected. Any ideas what am I doing wrong?

```python
from netbox_branching.models.branches import Branch

    def run(self, data, commit):

        branch = Branch.objects.all()
        for br in branch:
            self.log_debug(f"Branch name: {br.name}")
            self.log_info(f"Active: {br.is_active}")
```

Thanks!!

**Comment by mrmrcoleman** [comment]
Hey @RedLine89 did you figure this out?

**Comment by llamafilm** [comment, marked as answer]
@RedLine89 the code you wrote above works for me in a custom script.  It does not work in the nbshell, maybe that's how you were testing?  Another option I found that does the same thing:

```py
        from netbox_branching.utilities import get_active_branch
        branch = get_active_branch(self.request)
        if branch is not None:
            self.log_info(f"Branch: '{branch.name}' status: {branch.status}")
        else:
            self.log_info("Running in main")
```

**Reply by RedLine89** [reply]
Yeah, I think that after some updates in the branching plugin it now works. Thanks for confirming @llamafilm
